packages/espressocash_app/lib/features/tokens/services/token_service.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+import 'package:injectable/injectable.dart';
+import '../data/token_dto.dart';
+import '../data/token_repository.dart';
+
+@lazySingleton
+class TokenService {
+  TokenService(this._tokenRepository);
+
+  final TokenListRepository _tokenRepository;
+
+  Future<List<TokenDTO>> getAllTokens() => _tokenRepository.getAllTokens();
+
+  Stream<List<TokenDTO>> watchAllTokens() => _tokenRepository.watchAllTokens();
+
+  Future<void> insertToken(TokenDTO tokenDTO) =>
+      _tokenRepository.insertToken(tokenDTO);
+
+  Future<void> updateToken(TokenDTO tokenDTO) =>
+      _tokenRepository.updateToken(tokenDTO);
+
+  Future<void> deleteToken(TokenDTO tokenDTO) =>
+      _tokenRepository.deleteToken(tokenDTO);
+
+  Future<void> clearAllTokens() => _tokenRepository.clearAllTokens();
+
+  Future<void> initializeDatabaseWithJson(Map<String, dynamic> json) async {
```

Comment:
Where will this json come from? One of the main points is to get rid of having large json in memory.

It should rather be an assets csv file (versioned) that will be parsed on the app start whenever there's a new version.